### DISCUSSION TITLE
what happens if a scheduler process about to run a task at specified time dies?

### DISCUSSION THREAD

**Discussion by dk-github** [kagkarlsson/db-scheduler#431] [Q&A] [answered]
Hi, thank you for the lib!

I am just starting to evaluate it for our project, and RecurringTaskWithPersistentSchedule example suggests that for recurring tasks, schedules are persisted  in task_data blob combined with task's custom data (if any)?

What happens if it dies say 1 second before the next task execution would need to run/be scheduled - is it correct to assume it will get picked up by another instance (in multi-instance configuration)? 

If yes - what if it gets picked up say 1 second after the scheduled time - can it be configured to let it pass or execute late?

Finally, if an execution takes longer than the interval to next scheduled execution - what are the behaviour options, e.g. can one choose to start next one regardless, wait until previous one finishes, or skip this one?

Many thanks in advance,
Dmitry

**Comment by kagkarlsson** [comment, marked as answer]
> RecurringTaskWithPersistentSchedule example suggests that for recurring tasks, schedules are persisted in task_data blob combined with task's custom data (if any)?

Correct

> What happens if it dies say 1 second before the next task execution would need to run/be scheduled - is it correct to assume it will get picked up by another instance (in multi-instance configuration)?

Correct, it is not reserved for a specific instance

> If yes - what if it gets picked up say 1 second after the scheduled time - can it be configured to let it pass or execute late?

Not quite sure what you mean. The execution will run even if it should have run 1 hour ago. Schedule is only consulted when determining next execution time

> Finally, if an execution takes longer than the interval to next scheduled execution - what are the behaviour options, e.g. can one choose to start next one regardless, wait until previous one finishes, or skip this one?

Currently no options. It is skipped. When the current executions finishes, the next time is calculated based on Schedule and current time. 
For guaranteed execution, you might have a recurring task that spawns one-time tasks.
